File: app/src/main/java/com/jeeps/gamecollector/comparators/GameByHoursCompletionistComparator.java
package com.jeeps.gamecollector.comparators;

import com.jeeps.gamecollector.remaster.data.model.data.Game;
import com.jeeps.gamecollector.remaster.data.model.data.GameHoursStats;

import java.util.Comparator;

public class GameByHoursCompletionistComparator implements Comparator<Game> {
    private boolean desc;

    public GameByHoursCompletionistComparator(boolean desc) {
        this.desc = desc;
    }

    public GameByHoursCompletionistComparator() {
        this.desc = false;
    }

    @Override
    public int compare(Game game1, Game game2) {
        GameHoursStats gameHoursStats1 = game1.getGameHoursStats() != null ?
                game1.getGameHoursStats() : new GameHoursStats();

        GameHoursStats gameHoursStats2 = game2.getGameHoursStats() != null ?
                game2.getGameHoursStats() : new GameHoursStats();
        if (!desc)
            return Double.compare(
                    gameHoursStats1.getGameplayCompletionist(),
                    gameHoursStats2.getGameplayCompletionist());
        return Double.compare(
                gameHoursStats1.getGameplayCompletionist(),
                gameHoursStats2.getGameplayCompletionist()) * -1;
    }
}

File: app/src/main/java/com/jeeps/gamecollector/remaster/utils/comparators/GameByHoursCompletionistComparator.kt
package com.jeeps.gamecollector.remaster.utils.comparators

import com.jeeps.gamecollector.remaster.data.model.data.games.Game

class GameByHoursCompletionistComparator : Comparator<Game> {
    private val desc: Boolean

    constructor(desc: Boolean) {
        this.desc = desc
    }

    constructor() {
        desc = false
    }

    override fun compare(game1: Game, game2: Game): Int {
        val (_, _, hours1) = game1.gameHoursStats
        val (_, _, hours2) = game2.gameHoursStats
        return if (!desc)
            hours1.compareTo(hours2)
        else
            hours1.compareTo(hours2) * -1
    }
}